### PR TITLE
Fixes MySQL syntax for the table structure in CREATE

### LINKED ISSUE #1595: Creating an empty table in mysql
### Current behavior

While working on #1594, I noticed that creating a table without any column (unusual, isn't it? 😉 ),
the generated SQL query for MySQL is invalid:

```
  defmodule DuntMigration do
    use Ecto.Migration

    def change, do
      create table(:dummy)
    end
  end

=> CREATE TABLE "dummy" () ENGINE = INNODB;
```

### Expected behavior

It should generate `CREATE TABLE "dummy" ENGINE = INNODB;`.

I'll gladly do it, feel free to assign me.

### PR DESCRIPTION
**Pull request by xcambar**
fixes #1595

### CODE REVIEW

File: lib/ecto/adapters/mysql/connection.ex [hunk 1]
```diff
@@ -725,10 +732,13 @@ if Code.ensure_loaded?(Mariaex) do
       <<?`, name::binary, ?`>>
     end
 
-    defp assemble(list) do
+    defp assemble(list), do: assemble(list, " ")
+    defp assemble(list, joiner) do
       list
       |> List.flatten
-      |> Enum.join(" ")
+      |> Enum.reject(&is_nil(&1))
```
Comment (josevalim) [inline lib/ecto/adapters/mysql/connection.ex]: Can we drop the is_nil check? I think removing both nil and empty strings is a smell, we should only remove the empty string ones.

**Comment by josevalim** [comment]
@xcambar excellent, thank you! I have added one tiny comment.

**Comment by xcambar** [comment]
I removed the is_nil check by making sure that all the functions implied in CREATE statements return strings.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**